src/_pytest/_io/saferepr.py

```diff
@@ -1,8 +1,28 @@
+import pprint
 import sys
 
 from six.moves import reprlib
 
 
+def _safe_format_call(call, x, *args):
+    try:
+        # Try the vanilla repr and make sure that the result is a string
+        return call(x, *args)
+    except Exception:
```

Comment:
if the trackback is not used, just use `except Exception as e`

Reply:
Just to make it clear, I have not changed this part, I've moved it from `_callhelper`. I thought `exc_info` was used to get the class of an exception. Would it be ok to use `e.__class__.__name__` instead?

Reply:
@RonnyPfannschmidt Can you have a look, please? With `e.__class__.__name__` it looks like 

```
try:
    exc_name = e.__class__.__name__
except Exception:
    exc_name = "unknown"
try:
    exc_info = str(e)
except Exception:
    exc_info = "unknown"
```

Reply:
Makes sense.  I'd use `as exc` though.